tuxemon/npc.py

```diff
@@ -847,5 +847,87 @@ def alter_item_quantity(
 
         return success
 
+    def can_buy_item(self, item_slug: str, qty: int, unit_price: int) -> bool:
+        current_money = self.game_variables["money"]
+        if current_money:
+            return current_money >= qty * unit_price
+        # If no 'money' variable, must be an NPC. Always allow buying:
```

Comment:
Inconsistent with your other comment.

# Only players will have game_variables, so only update money
        # if it exists:

Reply:
I'm not sure what's inconsistent here, sorry - players have a custom 'money' variable, but NPC's don't. 

What I'm trying to explain with the comment is:
NPC's should not have a 'money' variable (for now) so it's not possible to do a check on how much money they have. So I need to always let the buy/sell transaction succeed, if the NPC has no 'money' variable. And that's fine, because they're an NPC, players won't be able to abuse this to get their buy transactions to always succeed. 

Player's override NPC, and always do have a 'money' variable when they're initialized, I've added that variable recently in tuxemon/player.py 
Therefore, if there IS a 'money' variable, then there is a player, and we need to
a) Do a check to see if there's enough money to buy something
b) Increase/decrease the money variable if the buy/sell transaction succeeds. 

I'm happy to rewrite the comment to make more sense (or maybe there is a bug there I haven't seen?)

I guess we COULD actually add a 'money' variable to the NPC's as well, and then you can only sell items to them if they have enough money? It would simplify the code, as both buyer and seller would always be NPC's, and always have a 'money' variable. 
I think there was a discussion about doing something like that. But for now I've only added the 'money' variable to the player. I'm happy to make that change if you think treating NPC and players the same would be better.

Reply:
```
        current_money = self.game_variables["money"]
        if current_money:
```
vs
```
        # Only players will have game_variables, so only update money or raise
        # exception if it exists:
        if self.game_variables:
```
The logic is wrong in one of these places. 

For NPCs, either game_variables is an empty dict, or it has "money" set to 0, which is it? If it's an empty dict then you will get a key error when you check "money".

Also, this will allow players to buy things if they have 0 money. Probably not what you wanted.

Reply:
> For NPCs, either game_variables is an empty dict, or it has "money" set to 0, which is it? If it's an empty dict then you will get a key error when you check "money".

> Also, this will allow players to buy things if they have 0 money. Probably not what you wanted.

Oops, you're right on both counts. This function was only tested on Players, not NPC's - and I'm not sure how I let it happen that you could buy things with 0 money. I'll fix these, and I might add unit tests around the can_buy and can_sell functions to ensure they're being tested properly. Thanks!